examples/basic/frameworks/haystack_deep_research_agent/tests/test_haystack_deep_research_agent.py

```
@@ -0,0 +1,94 @@
+import os
+import urllib.error
+import urllib.request
+from pathlib import Path
+import pytest
+
+
+def _opensearch_reachable(url: str) -> bool:
+    try:
+        with urllib.request.urlopen(
+            f"{url.rstrip('/')}/_cluster/health", timeout=1
+        ) as resp:
```

Ruff (0.13.1)

Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)